onboarding/source/amb_light_service.c

```diff
@@ -18,22 +28,96 @@ static void lightServiceTask(void * pvParameters);
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+    if(lightServiceTaskHandle == NULL){
+        xReturned = xTaskCreate(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                &lightServiceTaskHandle);
 
+    }
+
+    if(xReturned == pdFAIL){
+        return 0;
+    }
+
+    
+    
+    if(xLightServiceQueue == NULL){
+	    
+	xLightServiceQueue = xQueueCreate(LIGHT_SERVICE_QUEUE_SiZE, LIGHT_EVENT_SIZE);
+	    
+	if(xLightServiceQueue == NULL){
+        	return 0;
+	}
+    }
     /* USER CODE END */
     return 1;
 }
 
+uint16_t getLightSensorData(void)
+{
+
+	adcData_t lightData;
+ 	/** - Start Group1 ADC Conversion 
+ 	*     Select Channel 6 - Light Sensor for Conversion
+ 	*/
+	adcStartConversion(adcREG1, adcGROUP1);
+
+ 	/** - Wait for ADC Group1 conversion to complete */
+ 	while(!adcIsConversionComplete(adcREG1, adcGROUP1)); 
+
+	/** - Read the conversion result
+	*     The data contains the Light sensor data
+    */
+	adcGetData(adcREG1, adcGROUP1, &lightData);
+	
+	/** - Transmit the Conversion data to PC using SCI
+    */
+	return lightData.value;
+}
+
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    ASSERT(lightServiceTaskHandle != NULL);
+    light_event_t queueTask = NULL_EVENT;
+	
+        if (xQueueReceive(xLightServiceQueue, &queueTask, portMAX_DELAY) == pdTRUE) {
+		
+            if (queueTask == MEASURE_LIGHT){
+            
+            	uint16_t lightSensorData = getLightSensorData();
+            	char data[6] = {0};
+            	snprintf(data, 8, "%u\r\n", lightSensorData);
 
+            	sciPrintText(scilinREG, (unsigned char*)data, 5);
+        }  
+      
+    }
     /* USER CODE END */
 }
 
-uint8_t sendToLightServiceQueue(light_event_t *event) {
+/**
+ * @brief Send a light service to the queue
+ * @param light_event_t *event
+ **/
+uint8_t sendToLightServiceQueue(light_event_t *event) { 
     /* USER CODE BEGIN */
     // Send the event to the queue.
-    
+    if(event == NULL){
+        
+        return 0;
+    }
+    BaseType_t xReturned = xQueueSend(xLightServiceQueue, event, LIGHT_EVENT_SIZE);
+
+    if(xReturned == pdPASS){
```

Comment:
No need to change this, but another way this could've been done is with a ternary operator. It's basically a 1-line if-else statement. `return (xReturned == pdPASS) ? 1 : 0;`

Reply:
This is how it works -> `(condition) ? <value if true> : <value if false>`